rust_snuba/src/mutations/clickhouse.rs

```diff
@@ -62,16 +62,34 @@ impl ClickhouseWriter {
 
     async fn process_message(&self, message: &Message<MutationBatch>) -> anyhow::Result<()> {
         let queries = format_query(&self.table, message.payload());
+        self.run_queries(queries).await
+    }
+
+    async fn run_queries(&self, queries: Vec<Vec<u8>>) -> anyhow::Result<()> {
         let session_id = Uuid::new_v4().to_string();
 
+        let mut session_check = false;
+
         for query in queries {
-            self.client
-                .post(&self.url)
-                .query(&[("session_id", &session_id)])
-                .body(query)
-                .send()
-                .await?
-                .error_for_status()?;
+            let mut request = self.client.post(&self.url).query(&[
+                // ensure that the tables from CREATE TEMPORARY TABLE exist in subsequent
+                // queries
+                ("session_id", session_id.as_str()),
+                // ensure that HTTP status code is correct
+                // https://clickhouse.com/docs/en/interfaces/http#http_response_codes_caveats
+                // TODO: port to main consumer
+                ("wait_end_of_query", "1"),
+            ]);
+
+            if session_check {
+                // ensure that we are in a session. if some load balancer drops the querystring, we
+                // want to know
+                request = request.query(&[("session_check", "1")]);
+            }
```

Comment:
Will this `if` block prevent enabling session check for the 0th query request? `session_check` is set to `false` to start with.

Reply:
yep that's the intent. the session is upserted on the first request